src/solvers/dgsem_tree/dg_2d.jl

```diff
@@ -78,6 +78,26 @@ function create_cache(mesh::Union{TreeMesh{2}, StructuredMesh{2}, UnstructuredMe
   return (; fstar1_L_threaded, fstar1_R_threaded, fstar2_L_threaded, fstar2_R_threaded)
 end
 
+
+function create_cache(mesh::TreeMesh{2}, equations,
+                      volume_integral::VolumeIntegralStaggeredGrid, dg::DG, uEltype)
+
+  have_nonconservative_terms(equations) !== Val(false) && error("VolumeIntegralStaggeredGrid only defined for no nonconservative terms.")
+
+  cache = create_cache(mesh, equations,
+                       VolumeIntegralPureLGLFiniteVolume(volume_integral.volume_flux_fv),
+                       dg, uEltype)
+
+  A3dp1_x = Array{uEltype, 3}
+  A3dp1_y = Array{uEltype, 3}
+
+  fstaggered1_threaded = A3dp1_x[A3dp1_x(undef, nvariables(equations), nnodes(dg)+1, nnodes(dg)) for _ in 1:Threads.nthreads()]
+  fstaggered2_threaded = A3dp1_y[A3dp1_y(undef, nvariables(equations), nnodes(dg), nnodes(dg)+1) for _ in 1:Threads.nthreads()]
```

Comment:
These (and other variables) should be renamed as well once "staggered" is dropped from the volume integral name.

Reply:
Sounds reasonable. What other name do you think of @sloede @ranocha? For now, I renamed it just `fhat1` and `fhat2`, but I don't have a strong opinion on this.

Reply:
Let's decide the name of the basic stuff first